### PR TITLE
ENH: added benchmark for ITK ResampleImageFilter

### PR DESCRIPTION
**Pull request by dpshamonin**
The benchmark includes support for 1D/2D/3D images,
common interpolators (Linear, Nearest, BSpline),
common registration transforms and their combinations
(Affine, BSpline, Euler, Similarity, Translation, Identity)
and other options to fully benchmark Resample for registration tasks.

### CODE REVIEW

File: examples/Filtering/ResampleBenchmark.cxx [hunk 1]
```diff
@@ -0,0 +1,1365 @@
...
+        }
+      else
+        {
+          compositeTransform->AddTransform( similarityTransform );
+        }
+    }
+}
+
+//------------------------------------------------------------------------------
+// This helper function completely define the transform[s]
```
Comment (jhlegarreta) [inline examples/Filtering/ResampleBenchmark.cxx]: define: defines ?

File: examples/Filtering/ResampleBenchmark.cxx [hunk 2]
```diff
@@ -0,0 +1,1365 @@
...
+
+  // Translation
+  for ( std::size_t i = 0; i < Dimension; ++i )
+    {
+      parameters[i + Dimension] = -1.0 * ( ( i + 1.0 ) * transformIndex );
+    }
+}
+
+//------------------------------------------------------------------------------
+// This helper function completely set the single transform
```
Comment (jhlegarreta) [inline examples/Filtering/ResampleBenchmark.cxx]: set: sets ?

File: examples/Filtering/ResampleBenchmark.cxx [hunk 3]
```diff
@@ -0,0 +1,1365 @@
...
+// will be also executed with number of threads provided by option "-threads".
+//
+// The following ITK interpolations are supported in this benchmark:
+// itk::NearestNeighborInterpolateImageFunction
+// itk::LinearInterpolateImageFunction
+// itk::BSplineInterpolateImageFunction
+//
+// Command line argument "-i" "Linear" ["Nearest"] ["BSpline"]
+// Command line argument "-soi" controls spline order interpolator for the BSpline interpolator.
+// For "-i" "BSpline" the 0th - 5th order splines are supported, the default 3th.
```
[the remark this answers is not in the document]
Comment (dpshamonin) [reply examples/Filtering/ResampleBenchmark.cxx]: Thanks, I will fix the typos.

**Comment by dpshamonin** [comment]
@jhlegarreta 
Thanks for your kind words.  I will fix the typos you found and remove the dashed lines. We already using range based loops when they were suggested by Visual Assist. For the general ITK style, I just used the defined **.clang-format** from ITK itself, but I had to fix formatting afterwards, since Clang format had many issues formatting this code.

**Comment by phcerdan** [comment]
Great! :sparkles:

**Comment by jhlegarreta** [comment]
Thanks Dennis.

**Comment by thewtex** [comment]
:1st_place_medal: Thanks, @dpshamonin !

**Comment by phcerdan** [comment]
Great! this is awesome, but now the filtering label takes quite a bit of time.

```
Label Time Summary:
Core                       =   0.03 sec*proc (1 test)
Filtering                  = 285.63 sec*proc (65 tests)
PerformanceBenchmarking    =   0.19 sec*proc (6 tests)
Registration               =   8.38 sec*proc (3 tests)
Segmentation               =  27.36 sec*proc (4 tests)
```

Should we create a separate label and CMake option for for Resample? Opened #59

**Pull request merged**